docs/protocol/Multisig.md

```diff
@@ -0,0 +1,35 @@
+# Multisig
+
+Multisigs are confirmed on-chain by the `Multisig` contract. While the processor
+does create the multisig, and sign for it, making it irrelevant to the chain,
+confirming it on-chain solves the question of if the multisig was successfully
+created or not. If each processor simply asked all other processors for
+confirmation, votes lost to the network would create an inconsistent view. This
+is a form of the Byzantine Generals Problem, which can be resolved by placing
+votes within a BFT system.
+
+Confirmation requires all participants confirm the new set of keys. While this
+isn't BFT, despite the voting process being BFT, it avoids the scenario where
+only t (where t is the BFT threshold, as used in the t-of-n multisig)
+successfully generated shares, actually creating a t-of-t multisig in practice,
+which is not BFT. This does mean a single node can delay a churn, which is
+expected to be handled via a combination of slashing, and if necessary, removal.
+
+Validators are allowed to vote multiple times across sets of keys, with the
+first set to be confirmed becoming the set of keys for that validator set. These
+keys remain valid for the validator set until it is changed. If a validator set
+remains consistent despite the global validator set updating, their keys carry.
+If a validator set adds a new member, and then loses them, their historical keys
+are not reused.
```

Comment:
This is decently annoying to implement. We can't index from H(validators) to keys due to this last property, which was intended to let nodes move to another node without their history and obligations (at least, for this specific issue). We also can't index by global set ID, validator set index as the validator set may be unchanged across global set ID changes. Considering the multisig contract just tracks the most recent keys until updated, we could simply have the processor NOP in this case, handling it off chain? Reduce dev work and consensus obligations?